sql/parser: introduce ALTER VIRTUAL CLUSTER .. SET REPLICATION SOURCE cmd

msbutler opened the pull request:
Epic CRDB-47102

Release note (sql change): this patch adds the new ALTER VIRTUAL CLUSTER .. SET REPLICATION SOURCE cmd so a user can configure the producer jobs on the source cluster. Currently they can only configure the EXPIRATION WINDOW. This patch also removes the EXPIRATION WINDOW option from the consumer side cmd, ALTER VIRTUAL CLUSTER SET REPLICATION.

msbutler committed:
sql/parser: introduce ALTER VIRTUAL CLUSTER .. SET REPLICATION SOURCE cmd

Epic CRDB-47102

Release note (sql change): this patch adds the new ALTER VIRTUAL CLUSTER .. SET
REPLICATION SOURCE cmd so a user can configure the producer jobs on the source
cluster. Currently they can only configure the EXPIRATION WINDOW. This patch
also removes the EXPIRATION WINDOW option from the consumer side cmd, ALTER
VIRTUAL CLUSTER SET REPLICATION.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142501)
<!-- Reviewable:end -->

msbutler committed:
crosscluster/physical: process producer options seperately in ALTER TENANT

This patch refactors the implementation of ALTER VIRTUAL CLUSTER SET
REPLICATION SOURCE to process the options in a seperate function than the
consumer side ALTER VC SET REPLICATION. This refactor enables a future change
to auth the producer side changes with the REPLICATIONSOURCE priv and dest side
changes with the REPLICATIONDEST priv.

Epic: none

Release note: none

jeffswenson reviewed (approved):
LGTM

msbutler commented:
TFTR!

bors r=jeffswenson
